lib/block-supports/anchor.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Anchor block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the anchor block attribute for block types that support it.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_anchor_support( $block_type ) {
+	$has_anchor_support = _wp_array_get( $block_type->supports, array( 'anchor' ), true );
+	if ( ! $has_anchor_support ) {
+		return;
+	}
+
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
+	}
+
+	if ( ! array_key_exists( 'anchor', $block_type->attributes ) ) {
+		$block_type->attributes['anchor'] = array(
+			'type' => 'string',
+		);
+	}
+}
```

Comment:
This process would also apply to static blocks, which would add an extra comment as attributes.